vehicle-model-lifecycle/src/download_vspec.py

```diff
@@ -151,24 +169,46 @@ def main(app_manifest_dict: Dict[str, Any]) -> None:
                 f"Only up to one {FUNCTIONAL_INTERFACE_TYPE_KEY!r} supported!"
             )
         elif len(interfaces) == 1:
-            vspec_src = get_vehicle_signal_interface_src(interfaces[0])
+            vspec_src, unit_src_list = get_vehicle_signal_interface_src(interfaces[0])
+            if vspec_src == "":
+                vspec_src = "https://github.com/COVESA/vehicle_signal_specification/releases/download/v3.0/vss_rel_3.0.json"
+            if not unit_src_list:
+                unit_src_list = [
+                    "https://github.com/COVESA/vehicle_signal_specification/blob/v4.0/spec/units.yaml",
+                ]
         else:
             # FIXME: Fallback solution in case an app does not provide a VSS
             #        file. Code path can be removed once we have a dependency
             #        resolver for our runtimes.
-            vspec_src = os.path.join(
-                get_package_path(), "vehicle-model-lifecycle", "vspec.json"
-            )
+            vspec_src = "https://github.com/COVESA/vehicle_signal_specification/releases/download/v3.0/vss_rel_3.0.json"
+            unit_src_list = [
+                "https://github.com/COVESA/vehicle_signal_specification/blob/v4.0/spec/units.yaml",
+            ]
 
     local_vspec_path = os.path.join(get_workspace_dir(), os.path.normpath(vspec_src))
 
     if is_uri(vspec_src):
         local_vspec_path = os.path.join(get_project_cache_dir(), "vspec.json")
         download_file(vspec_src, local_vspec_path)
 
-    vspec_src = local_vspec_path
+    id = 0
+    list = []
+    for unit_src in unit_src_list:
+        if is_uri(unit_src):
+            local_unit_path = os.path.join(get_project_cache_dir(), f"units_{id}.yaml")
+            # since there is no release for units file 4.0 we need to download from blob and this is different than downloading from release
+            if unit_src.find("blob") != -1:
+                download_unit_file(unit_src, local_unit_path)
+            else:
+                download_file(unit_src, local_unit_path)
+            list.append(local_unit_path)
+            id += 1
```

Comment:
please create a function "get_vehicle_signal_interface_units_files" and do all that (including line 208+209) and do:

unit_src_list = get_vehicle_signal_interface_units_files(unit_src_list)